snuba/pipeline/query_pipeline.py

```diff
@@ -35,10 +45,26 @@ def execute(self) -> QueryResult:
 
 class QueryPipelineBuilder(ABC):
```

Comment:
I think this is unlikely to stay the way it is once we have the composite query in the picture.
There will most likely be an entity builder (that provides an execution and a processing pipeline for the single entity query and provided by an entity) and a dataset builder that provides an execution and a processing pipeline for requests. 
Though, how this finally look like will strongly depend on the pipeline delegator which is being built here: https://github.com/getsentry/snuba/pull/1465 So I will let that PR be finalized before making major changes to the builder.